provisioner/terraform/provision.go

```diff
@@ -87,12 +110,21 @@ func (t *server) Provision(stream proto.DRPCProvisioner_ProvisionStream) error {
 		})
 	}
 
-	t.logger.Debug(shutdown, "running initialization")
-	err = e.init(stream.Context(), logr)
+	s.logger.Debug(ctx, "running initialization")
+	err = e.init(ctx, killCtx, logr)
 	if err != nil {
+		if ctx.Err() != nil {
+			return stream.Send(&proto.Provision_Response{
+				Type: &proto.Provision_Response_Complete{
+					Complete: &proto.Provision_Complete{
+						Error: err.Error(),
+					},
+				},
+			})
+		}
```

Comment:
@kylecarbs yay or nay on this change? I'm thinking it'd be OK to return provision complete on cancellation, similar to `plan` and `apply` below.

Reply:
I'm good with it!